src/content/getusermedia/gum/js/main.js

```diff
@@ -5,47 +5,44 @@
  *  that can be found in the LICENSE file in the root of the source
  *  tree.
  */
-
 'use strict';
 
-var errorElement = document.querySelector('#errorMsg');
-var video = document.querySelector('video');
-
 // Put variables in global scope to make them available to the browser console.
-var constraints = window.constraints = {
+const constraints = window.constraints = {
   audio: false,
   video: true
 };
 
 function handleSuccess(stream) {
-  var videoTracks = stream.getVideoTracks();
+  const video = document.querySelector('video');
+  const videoTracks = stream.getVideoTracks();
   console.log('Got stream with constraints:', constraints);
-  console.log('Using video device: ' + videoTracks[0].label);
-  stream.oninactive = function() {
-    console.log('Stream inactive');
-  };
+  console.log(`Using video device: ${videoTracks[0].label}`);
   window.stream = stream; // make variable available to browser console
   video.srcObject = stream;
 }
 
 function handleError(error) {
   if (error.name === 'ConstraintNotSatisfiedError') {
-    errorMsg('The resolution ' + constraints.video.width.exact + 'x' +
-        constraints.video.width.exact + ' px is not supported by your device.');
+    let v = constraints.video;
+    errorMsg(`The resolution ${v.width.exact}x${v.height.exact} px is not supported by your device.`);
   } else if (error.name === 'PermissionDeniedError') {
     errorMsg('Permissions have not been granted to use your camera and ' +
       'microphone, you need to allow the page access to your devices in ' +
       'order for the demo to work.');
   }
-  errorMsg('getUserMedia error: ' + error.name, error);
+  errorMsg(`getUserMedia error: ${error.name}`, error);
 }
 
 function errorMsg(msg, error) {
-  errorElement.innerHTML += '<p>' + msg + '</p>';
+  const errorElement = document.querySelector('#errorMsg');
+  errorElement.innerHTML += `<p>${msg}</p>`;
   if (typeof error !== 'undefined') {
     console.error(error);
   }
 }
 
-navigator.mediaDevices.getUserMedia(constraints).
-    then(handleSuccess).catch(handleError);
+navigator.mediaDevices
```

Comment:
Can we async/await and put the success/error code inline instead of use-once helper functions?

```
(async () => {
  try {
    const stream = await navigator.mediaDevices.getUserMedia(constraints);
    ... do whatever handleSuccess(stream) does ...
  } catch (e) {
    ... do whatever handleError(e); does ...
  }
})();
```